src/api/firebase.js

```diff
@@ -71,8 +73,44 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 
 export async function updateItem(listId, docId, itemData) {
 	const docRef = doc(db, listId, docId);
+	const { isChecked, totalPurchases, dateLastPurchased, dateNextPurchased } =
+		itemData;
+	let data;
 
-	await updateDoc(docRef, itemData);
+	const currentDate = new Date();
+	const oldDateLastPurchased = dateLastPurchased
+		? dateLastPurchased.toDate()
+		: currentDate;
```

Comment:
Here we set the `currentDate` to `oldDateLastPurchased` if it's the first time we're buying the item (`dateLastPurchased` doesn't exist)